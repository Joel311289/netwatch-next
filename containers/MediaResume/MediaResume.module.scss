$card-height: 240px;

.wrapper {
  position: relative;
  display: block;
  height: $card-height;
}

.body {
  @include absolute(top 0 left 0);
  @include box(100%, $card-height);

  background: linear-gradient(0deg, rgb(20 20 20 / 80%), transparent 1500px);
  color: var(--color-button-text);
  z-index: 1;
}

.buttonBack {
  padding: 0 10px 0 5px !important;
}
