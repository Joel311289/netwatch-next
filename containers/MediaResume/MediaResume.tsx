import { useTranslation } from 'next-i18next';
import { FiArrowLeft } from 'react-icons/fi';
import Link from 'next/link';
import { ElementHTML, MediaImageRatio, MovieDetail, TVDetail } from '@/types';
import { Heading, Text } from '@/components/typography';
import { Image } from '@/components/media';
import { Container, Space } from '@/components/layout';
import { Button } from '@/components/forms';
import { Card } from '@/components/display';
import { classes } from '@/utils/helpers';
import styles from '@/containers/MediaResume/MediaResume.module.scss';

export type MediaResumeProps = typeof defaultProps &
  ElementHTML & {
    media?: MovieDetail | TVDetail;
  };

const defaultProps = {
  media: {}
};

const MediaResume = ({ media }: MediaResumeProps) => {
  const { t } = useTranslation();

  return (
    <div className={classes(styles.wrapper, 'full', 'dark')}>
      <Image alt="backdrop" src={media.backdrop || ''} width="100%" />

      <Space justify="center" className={styles.body} direction="column" gap={2}>
        <Container>
          <Space gap={25} direction="row">
            <Card>
              <Image
                alt="image"
                src={media.image}
                width={110}
                ratio={MediaImageRatio.POSTER}
                lazy
              />
            </Card>

            <Space gap={10} direction="column">
              <Heading level={3}>{media.name}</Heading>
              <Text size="sm">{[media.date, media.duration].compact().join(' - ')}</Text>

              <Link
                href={{
                  pathname: '/[type]/[id]',
                  query: { type: media.type, id: media.id }
                }}>
                <a>
                  <Button className={styles.buttonBack} clear>
                    <FiArrowLeft />
                    {t('main.go.back.button')}
                  </Button>
                </a>
              </Link>
            </Space>
          </Space>
        </Container>
      </Space>
    </div>
  );
};

MediaResume.defaultProps = defaultProps;

export default MediaResume;
