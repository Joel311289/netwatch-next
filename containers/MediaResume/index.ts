export { default as MediaResume } from '@/containers/MediaResume/MediaResume';
