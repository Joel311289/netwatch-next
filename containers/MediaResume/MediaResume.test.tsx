import { render } from '@testing-library/react';
import MediaResume from '@/containers/MediaResume/MediaResume';

describe('Tests MediaResume component', () => {
  test('renders component correctly', () => {
    const { container } = render(<MediaResume>Test</MediaResume>);

    expect(container).toMatchSnapshot();
  });
});
