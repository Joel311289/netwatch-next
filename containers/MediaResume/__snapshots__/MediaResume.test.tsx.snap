// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests MediaResume component renders component correctly 1`] = `
<div>
  <div
    class="wrapper full dark"
  >
    <div
      aria-label="backdrop"
      class="wrapper"
      role="img"
    >
      <div
        class="content"
      />
      <style>
        
        .wrapper {
          width: 100%;
          height: 100%;
          min-width: 100%;
        }
        .wrapper::before {
          padding-top: 0;
        }
      
      </style>
    </div>
    <div
      class="wrapper body"
    >
      <div
        class="wrapper"
      >
        <div
          class="wrapper"
        >
          <div
            class="wrapper"
          >
            <div
              aria-label="image"
              class="wrapper"
              role="img"
            >
              <div
                class="content"
              />
              <style>
                
        .wrapper {
          width: 110px;
          height: 165px;
          min-width: 110px;
        }
        .wrapper::before {
          padding-top: 150%;
        }
      
              </style>
            </div>
          </div>
          <div
            class="wrapper"
          >
            <h3
              class="wrapper"
            />
            <span
              class="wrapper sm"
            >
              undefined - undefined
              <style>
                
        .wrapper {
          
          
          
        }
      
              </style>
            </span>
            <a
              href="//"
            >
              <button
                class="wrapper medium clear buttonBack"
              >
                <svg
                  fill="none"
                  height="1em"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  stroke-width="2"
                  viewBox="0 0 24 24"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <line
                    x1="19"
                    x2="5"
                    y1="12"
                    y2="12"
                  />
                  <polyline
                    points="12 19 5 12 12 5"
                  />
                </svg>
                main.go.back.button
              </button>
            </a>
            <style>
              
        .wrapper {
          display: flex;
          flex-direction: column;
          flex-wrap: nowrap;
          gap: 10px;
          
          
        }
      
            </style>
          </div>
          <style>
            
        .wrapper {
          display: flex;
          flex-direction: row;
          flex-wrap: nowrap;
          gap: 25px;
          
          
        }
      
          </style>
        </div>
      </div>
      <style>
        
        .wrapper {
          display: flex;
          flex-direction: column;
          flex-wrap: nowrap;
          gap: 2px;
          justify-content: center;
          
        }
      
      </style>
    </div>
  </div>
</div>
`;
